Dockerfile

```diff
@@ -6,7 +6,7 @@ ARG XX_VERSION=1.2.1
 ARG DOCKER_VERSION=24.0.2
 ARG GOTESTSUM_VERSION=v1.9.0
 ARG REGISTRY_VERSION=2.8.0
-ARG BUILDKIT_VERSION=v0.11.6
+ARG BUILDKIT_VERSION=master@sha256:58b3d496584c7a33dd129fb4f6dec7828c4f31aa7808a5be1c9835f0fbc33bfe
```

Comment:
Let's not update this. Vendor-only if not a release.

Reply:
See https://github.com/docker/buildx/pull/2023#discussion_r1306906177, I think this is necessary.

We could cherry-pick it to v0.12 branch and release? Otherwise, I don't see a trivial way to solve this - unless we detect the presence of the flag for the tests (@crazy-max).

Reply:
Yes it's necessary as we are not synced between vendoring adding the `otel-socket-path` flag that is not available in a current buildkit release. We could conditionally set this flag only if available in buildkitd help output like https://github.com/moby/buildkit/compare/master...crazy-max:buildkit:integration-check-flag but I don't think it's worth it.

Reply:
@jedevc With https://github.com/moby/buildkit/pull/4280 this is not necessary anymore. See https://github.com/docker/buildx/pull/2060.

Reply:
Pulled the changes in, done :)